packages/chips/ChipSet.js

```diff
@@ -41,6 +54,20 @@ export default class ChipSet extends Component {
     };
   }
 
+  handleSelect = (chipId) => {
+    if (this.props.filter || this.props.choice) {
+      this.foundation_.toggleSelect(chipId);
+    }
+    this.props.handleSelect(this.foundation_.getSelectedChipIds());
```

Comment:
`MDCChipSetFoundation` already keeps a list of the select chip ids internally, so we can use that instead of keeping one in the React ChipSet. This means that `adapter.setSelected` won't be used.

Talked offline with @moog16: an alternative we considered is instead of having the user pass a `handleSelect(allSelectedChipIds)` callback, have the user pass a `setSelected(chipId, selected)` callback which will be called from `adapter.setSelected` (within `foundation.toggleSelect`).